taichi/analysis/build_cfg.cpp

```diff
@@ -130,26 +130,20 @@ class CFGBuilder : public IRVisitor {
   void visit(OffloadedStmt *stmt) override {
     if (stmt->has_body()) {
       auto before_offload = new_node(-1);
-      if (stmt->task_type == OffloadedStmt::TaskType::struct_for ||
-          stmt->task_type == OffloadedStmt::TaskType::range_for) {
-        visit_loop(stmt->body.get(), before_offload, false);
-      } else {
-        TI_ASSERT(stmt->task_type == OffloadedStmt::TaskType::serial);
-        int offload_stmt_id = current_stmt_id;
-        auto block_begin_index = graph->size();
-        stmt->body->accept(this);
-        prev_nodes.push_back(graph->back());
-        // Container statements don't belong to any CFGNodes.
-        begin_location = offload_stmt_id + 1;
-        CFGNode::add_edge(before_offload,
-                          graph->nodes[block_begin_index].get());
-      }
+      int offload_stmt_id = current_stmt_id;
+      auto block_begin_index = graph->size();
+      stmt->body->accept(this);
+      prev_nodes.push_back(graph->back());
+      // Container statements don't belong to any CFGNodes.
+      begin_location = offload_stmt_id + 1;
+      CFGNode::add_edge(before_offload, graph->nodes[block_begin_index].get());
```

Comment:
I have a question here: are we building a CFG for the body of the OffloadedStmt here that is completely unconnected to other parts of the graph? (If this is the case everything will be easier to understand.)

Reply:
No. For example, in this case, 
```
$0 = offload list gen ...
$1 = offload {
  ...
}
$2 = offload {
  ...
}
$3 = offload {
  ...
}
```
the CFG will be
```
Node 0 (the start node of CFG): $0, next={Node 1}
Node 1: the first node of $1's body, next=...
...
Node a: the last node of $1's body, next={Node a+1}
Node a+1: the first node of $2's body, next=...
...
Node b: the last node of $2's body, next={Node b+1}
Node b+1: the first node of $3's body, next=...
...
Node c: the last node of $3's body
```

Reply:
(And so I need to check if the stored data is visible to the load_stmt before doing store forwarding...)

Reply:
What about making each OffloadedStmt's CFG unconnected, and make `ControlFlowGraph::start_node` an `std::vector` rather than a `const int`?

Reply:
We can perform some cross-offload analyses and optimizations if we make every node connected and if we don't consider each global ptr as a source of data... 

For example:
```
$1 : offload {
  $2 = global ptr a
  $3 = arg load [0]
  $4 : global store $2 <- $3
}
$5 : offload {
  $6 : global ptr a
  $7 : arg load [0]
  $8 : if ... {
    $9 : global store $6 <- $7
  }
  $10 : global load $6 (can be replaced with $7 if we don't consider $6 as a source of data))
}

Reply:
I think we can basically stick to what you have now. If we should do cross-offload optimization here needs more discussions.